### ISSUE TITLE
Change `enum` syntax to more closely match `struct` syntax

### ISSUE THREAD

**Issue by Possseidon** [Possseidon/parser-playground#8]
What it currently looks like:

```rs
enum Example {
    Tokens: [
        Ident,
        Number,
    ]
    Nodes: [
        Node1,
        Node2,
    ]
}
```

Things to note on the current design:

- Variant names and underlying token/linked node are automatically the same
  - Great, since it would usually only lead to confusion to give a variant a different name
- Tokens and Nodes are separated
  - Since the grammar doesn't care about the order of alternations this is not an issue
  - It's probably easier to read if they're separated anyway, so enforcing this is not a bad thing imo
- It does *not* use the `[Token]` and `(Node)` syntax, that `struct`s are using
  - The main reason for this syntax was to allow for the optional/repetition (like `(Node?)`) which doesn't make sense for alternations
  - But I still think it would be more readable to use the same syntax and ditch the verbose `Tokens: []` and `Nodes: []` headers
  - This would also get rid of a level of indentation, which is always welcome

To summarize, the current design is great, but I want to change the syntax to the following:

```rs
enum Example {
    [Ident],
    [Number],
    (Node1),
    (Node2),
}
```

To make it more readable (but also keep the implementation simple) tokens are still required to come before nodes.

Getting rid of the header even simplifies the implementation, since I don't need double unwrapping `$( $( $Field )* )?` anymore.

I also considered two other options:

- Just putting some separator between tokens and nodes and ditch the parens/brackets entirely
- Putting one set of parens/brackets around the entire thing

... but mirroring the `struct` field syntax completely just feels the most natural to me.
